profiler/profiler.go

```diff
@@ -353,12 +362,34 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 		for _, prof := range completed {
 			bat.addProfile(prof)
 		}
-		p.enqueueUpload(bat)
+
+		// Wait until the next profiling period starts or the profiler is stopped.
 		select {
 		case <-ticker:
+			// Usually ticker triggers right away because the non-CPU profiles cause
+			// the wg.Wait above to sleep until the end of the profiling period.
+			// Edge case: If only the CPU profile is enabled, and the cpu duration is
+			// is less than the configured profiling period, the ticker will block
+			// until the end of the profiling period.
 		case <-p.exit:
 			return
 		}
+
+		// Include endpoint hits from tracer in profile `event.json`.
+		// Also reset the counters for the next profile period.
+		bat.endpointCounts = endpointCounter.GetAndReset()
+		// Record the end time of the profile.
+		// This is used by the backend to upscale the endpoint counts if the cpu
+		// duration is less than the profile duration. The formula is:
+		//
+		// factor = (end - start) / cpuDuration
+		// counts = counts * factor
+		//
+		// The default configuration of the profiler (cpu duration = profiling
+		// period) results in a factor of 1.
+		bat.end = time.Now()
```

Comment:
Assuming somebody sets the CPU duration to less than the profile period, bat.end now means something different than it did before. From Nikolay's approval, I take it that this is okay from the backend perspective?

Reply:
Yes, this was discussed with @mar-kolya and he reached out to @pmbauer who previously added a comment on `bat.end` behavior in the code. They both agreed that this can and should be changed.